Source/WarpX.cpp

```diff
@@ -1520,7 +1520,8 @@ WarpX::AllocLevelMFs (int lev, const BoxArray& ba, const DistributionMapping& dm
     }
 #endif
 
-    bool deposit_charge = do_dive_cleaning || (plot_rho && do_back_transformed_diagnostics);
+    bool deposit_charge = do_dive_cleaning || (plot_rho && do_back_transformed_diagnostics)
+                          || (do_electrostatic != ElectrostaticSolverAlgo::None);
```

Comment:
As the code is now, this should be `do_electrostatic == ElectrostaticSolverAlgo::LabFrame` since the other option still uses its own local copy of rho.